test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/FormActionTagHelperTest.cs

```diff
@@ -0,0 +1,387 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.Routing;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using Microsoft.AspNetCore.Routing;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.TagHelpers
+{
+    public class FormActionTagHelperTest
+    {
+        [Fact]
+        public async Task ProcessAsync_GeneratesExpectedOutput()
```

Comment:
Ah, so I see how the rest of the tests in this method don't technically ensure expected output. Does this single test really validate the various output scenarios? I only ask because you took this route vs the functional test route.